### PR TITLE
Add maximum projection module

### PR DESCRIPTION
**Pull request by jenna-tomkinson**
# Add maximum projection module

In this PR, the maximum projection module is added to generate a 2D representative image per well and site. A check notebook is also included to make sure that there is complete equal number of channels across all image sets per z-slice.

### CODE REVIEW

File: 1.max_projection/scripts/0.check_incomplete_sets.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    # Extract well and site (e.g., 'C10-1' -> well='C10', site='1')
+    well_site = parts[0]
+    well = well_site.split("-")[0]  # Extract well (e.g., 'C10')
+    site = well_site.split("-")[1]  # Extract site (e.g., '1')
+
+    # Extract Z-slice identifier (e.g., ZS000)
+    z_slice = parts[2]  # Example: 'ZS000'
+
+    # Create a unique identifier for the image set (well, site, Z-slice)
+    set_id = f"{well}_site{site}_{z_slice}"
```
Comment (MikeLippincott) [inline 1.max_projection/scripts/0.check_incomplete_sets.py:63]: Nice documentation here!

File: 1.max_projection/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Generate max-projection image sets
+
+In this module, we use CellProfiler to group the z-slices together per well and site and use maximum projection (taking the highest intensity per pixel) for each channel to create one representative image set.
+
+There are 104 unique wells and sites (1 removed due to incomplete number of channels) with 5 channels each, so we took >20,000 images and reduced it down to 520 images (104 image sets * 5 channels).
+It took approximately **one hour** to generate the maximum projected image sets.
+We are using a Linux-based machine running Pop_OS! LTS 22.04 with an AMD Ryzen 7 3700X 8-Core Processor. There is a total of 16 CPUs with 125 GB of MEM.
```
[the remark this answers is not in the document]
Comment (jenna-tomkinson) [reply 1.max_projection/README.md:7]: Thanks! :D Honestly it helps me for if I have to rerun stuff to know how much time to prioritize.

File: 1.max_projection/scripts/0.check_incomplete_sets.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+from collections import defaultdict
+
+
+# ## Set paths and variables
+
+# In[2]:
+
+
+# Define the path to your directory containing the images
+image_dir = Path("/media/18tbdrive/GFF_organoid_data/Cell Painting-NF0014 Thawed3-Pilot Drug Screening/NF0014-Thawed 3 (Raw image files)-Combined/NF0014")
```
[the remark this answers is not in the document]
Comment (jenna-tomkinson) [reply 1.max_projection/scripts/0.check_incomplete_sets.py:28]: I know :( I am not a fan of it also, but I don't particularly want to have to mess with this data more than I have to, given the leaps we have to take currently.

I will leave it for now, keeping in mind the risk of downstream issues; I believe that we should be good now that the images are max-projected.

File: 1.max_projection/scripts/0.check_incomplete_sets.py (hunk 1 above)
Comment (jenna-tomkinson) [reply 1.max_projection/scripts/0.check_incomplete_sets.py:63]: Thank you! :D

**Comment by jenna-tomkinson** [comment]
@MikeLippincott Thank you for the review! I am merging now!

**Pull request merged**